### PR TITLE
Move preview URL logic to view

### PR DESCRIPTION
**Pull request by dextermilo**
### Issues
1. Preview URLs were broken related to a misconfiguration of the env variable
2. Preview URLs were generated and stored in the DB during bundle processing which means they would not update with changes to the env variables.

This PR moves the URL related variables into the view. This has a side effect of requiring auth to SFDC to determine the sandbox URLs.

### CODE REVIEW

File: config/settings/production.py [hunk 1]
```diff
@@ -139,7 +139,7 @@
     "SALESFORCE_ARTICLE_URL_PATH_PREFIX", default="/articles/Resource/"
 )
 SALESFORCE_ARTICLE_PREVIEW_URL_PATH_PREFIX = env(
-    'SALESFORCE_ARTICLE_URL_PATH_PREFIX',
+    'SALESFORCE_ARTICLE_PREVIEW_URL_PATH_PREFIX',
```
Comment (prescod) [inline config/settings/production.py]: Are there other places this change should be made until we clean up the settings?

Comment (dextermilo) [reply config/settings/production.py]: Yes. https://github.com/SFDO-Tooling/sfdoc/pull/235/commits/c6c0459b6bd1cce1645156dfc6d367d56480d637

File: sfdoc/publish/views.py [hunk 2]
```diff
@@ -138,7 +140,16 @@ def review(request, pk):
         return HttpResponseRedirect('../')
     else:
         form = PublishToProductionForm()
+    try:
+        base_url = get_community_base_url()
+    except Exception as error:
+        print(error)
+        print('Error getting base URL')
```
Comment (davisagli) [inline sfdoc/publish/views.py]: Can you explain the error handling here? I think this is going to print to somewhere we'll never see, and then break with a NameError on line 149. Would be easier to track down if you don't swallow the exception.

Comment (dextermilo) [reply sfdoc/publish/views.py]: @davisagli this wasn't ready for primetime

**Comment by dextermilo** [comment]
Sorry @davisagli @prescod, I made changes after requesting review :-\

1) DB column for preview_url has been removed, property added in its place, migration added
2) Review view has been refactored for testability
3) Basic test for view added

File: sfdoc/publish/views.py [hunk 3]
```diff
@@ -138,18 +140,31 @@ def review(request, pk):
...
+    assert base_url is not None
+
+    def article_for_view(item):
+        return {
+            'preview_url': '{}{}'.format(base_url, item.preview_url),
+            'url_name': item.url_name,
+        }
+
+    def get_articles_for_view(collection):
+        return list(map(article_for_view, collection))
```
Comment (davisagli) [inline sfdoc/publish/views.py]: Consider:

    return [{
        "preview_url": f"{base_url}{article.preview_url}",
        "url_name": article.url_name,
    } for article in collection]

Comment (dextermilo) [reply sfdoc/publish/views.py]: I write too much javascript.

Resolved by https://github.com/SFDO-Tooling/sfdoc/pull/235/commits/9e1862c180f8cfba60784e4ef50bf2cc51457346

**Pull request merged**